src/module.js

```diff
@@ -124,106 +130,151 @@ function createInitialHalsteadItemState () {
+         if (Object.prototype.hasOwnProperty(identifier)) {
+         // Avoid clashes with built-in property names.
+         incrementDistinctHalsteadItems(baseReport, metric, '_' + identifier);
+         } else if (isHalsteadMetricDistinct(baseReport, metric, identifier)) {
+         recordDistinctHalsteadMetric(baseReport, metric, identifier);
+         incrementHalsteadMetric(baseReport, metric, 'distinct');
+         }
+         }
+
+         function isHalsteadMetricDistinct (baseReport, metric, identifier) {
+         return baseReport.halstead[metric].identifiers.indexOf(identifier) === -1;
+         }
+
+         // first if could moved up
+         // isHalsteadMetricDistinct(baseReport, metric, identifier) return to negate with return
+         */
+        if (actualReport.halstead[type].identifiers.indexOf(identifier) !== -1) {
+            return;
+        }
 
-function isHalsteadMetricDistinct (baseReport, metric, identifier) {
-    return baseReport.halstead[metric].identifiers.indexOf(identifier) === -1;
-}
+        // recordDistinctHalsteadMetric(actualReport, type, saveIdentifier);
+        actualReport.halstead[type].identifiers.push(identifier);
 
-function recordDistinctHalsteadMetric (baseReport, metric, identifier) {
-    baseReport.halstead[metric].identifiers.push(identifier);
-}
 
-function incrementHalsteadMetric (baseReport, metric, type) {
-    if (baseReport) {
-        baseReport.halstead[metric][type] += 1;
-    }
-}
+        // incrementHalsteadMetric(actualReport, type, 'distinct');
+        actualReport.halstead[type].distinct += 1;
+    };
 
-function incrementTotalHalsteadItems (baseReport, metric) {
-    incrementHalsteadMetric(baseReport, metric, 'total');
+        return function(node, operatorsOrOperands, currentReport) {
+            if (!Array.isArray(operatorsOrOperands)) {
+                return;
+            }
+            /**
+             * oooItem is the short variant of operatorsOrOperandsItem
+             */
+            operatorsOrOperands.forEach(function (oooItem) {
+                /**
+                 var identifier;
+
+                 if (check.function(s.identifier)) {
+                    identifier = s.identifier(node);
+                } else {
+                    identifier = s.identifier;
+                }
+                 refactored to
+                 */
+                var identifier = check.function(oooItem.identifier) ? oooItem.identifier(node) : oooItem.identifier;
+                //identifier = Object.prototype.hasOwnProperty(identifier) ? '_' + identifier : identifier;
+                /*
+                 if (check.function(s.filter) === false || s.filter(node) === true) {
+                 halsteadItemEncountered(currentReport, metric, identifier);
+                 }
+
+                 !a || b -> a && !b
+                 */
+                if (check.function(oooItem.filter) && !oooItem.filter(node)) {
+                    return;
```

Comment:
Please avoid using return statements purely for control flow. It's usually clearer (although more cluttered) to use nested conditional blocks instead.